extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/test/java/io/quarkus/resteasy/reactive/links/deployment/TestRecordWithIdAndPersistenceId.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.resteasy.reactive.links.deployment;
+
+import io.quarkus.resteasy.reactive.links.deployment.persistence.Id;
```

Comment:
😂

Reply:
I need this to be able to test the Id annotation scanning without having to depend on JPA. That is why I rather have the scan for Id removed, but I do nit want to introduce breaking changes.

Reply:
That is also why I introduced the RestLinkId annotation. It is allowed only once in a class. The JPA spec allows for multiple Id field which would the self link to break.